idaes/generic_models/properties/core/eos/ceos.py

```diff
@@ -349,7 +349,92 @@ def dens_mol_phase(b, p):
         else:
             raise PropertyNotSupportedError(_invalid_phase_msg(b.name, p))
 
-    # TODO: Need to add functions to calculate cp and cv
+
+    @staticmethod
+    def cp_mol_phase(blk, p):
+        pobj = blk.params.get_phase(p)
+        if not pobj.is_vapor_phase():
+            raise PropertyNotSupportedError(_invalid_phase_msg(blk.name, p))
+
+        cname = pobj._cubic_type.name
+        am = getattr(blk, cname+"_am")[p]
+        bm = getattr(blk, cname+"_bm")[p]
+        A = getattr(blk, cname+"_A")[p]
+        B = getattr(blk, cname+"_B")[p]
+        fw = getattr(blk, cname+"_fw")
+        kappa = getattr(blk.params, cname+"_kappa")
+        dadT = getattr(blk, cname+"_dadT")[p]
+        Z = blk.compress_fact_phase[p]
+
+        EoS_u = EoS_param[pobj._cubic_type]['u']
+        EoS_w = EoS_param[pobj._cubic_type]['w']
+        omegaA = EoS_param[pobj._cubic_type]['omegaA'] 
+        EoS_p = sqrt(EoS_u**2 - 4*EoS_w)
+
+        d2adT2 = - (0.5 / blk.temperature) * dadT + \
+                 ((Cubic.gas_constant(blk)**2 * omegaA) / (2*blk.temperature)) * (
+                     sum(sum(
+                         blk.mole_frac_phase_comp[p, i] * blk.mole_frac_phase_comp[p, j]
+                         * (1 - kappa[i, j]) * fw[i] * fw[j] * sqrt(
+                             (blk.params.get_component(i).temperature_crit * blk.params.get_component(j).temperature_crit) /
+                             (blk.params.get_component(i).pressure_crit * blk.params.get_component(j).pressure_crit)
+                         )
+                         for j in blk.components_in_phase(p))
+                         for i in blk.components_in_phase(p))
+                 )
+
+        dBdT = -B / blk.pressure
+        dAdT = (A/am) * dadT - (2*A/blk.temperature)
+        K2 = (EoS_u - 1) * B - 1
+        K3 = A - EoS_u * B - (EoS_u - EoS_w) * B**2
+        K4 = - (A * B + EoS_w * B**2 + EoS_w * B**3)
+        dK2dT = (EoS_u - 1) * dBdT
+        dK3dT = dAdT - EoS_u * dBdT - 2 * (EoS_u - EoS_w) * B * dBdT
+        dK4dT = - (A * dBdT + B * dAdT + 2 * EoS_w * B * dBdT + 3 * EoS_w * B**2 * dBdT) 
+        dZdT = - (Z**2 * dK2dT + Z*dK3dT + dK4dT) / (3*Z**2 + 2*K2*Z + K3)
+
+        expression1 = 2 * Z + (EoS_u + EoS_p) * B
+        expression2 = 2 * Z + (EoS_u - EoS_p) * B
+        expression3 = (2 * dZdT + (EoS_u - EoS_p) * dBdT) / expression2
+        expression4 = (2 * dZdT + (EoS_u + EoS_p) * dBdT) / expression1
+
+
+        # Derived from equation on pg. 120 in Properties of Gases and Liquids
```

Comment:
This comment is not correct for this method - can you give a reference to the source you used instead?